### PR TITLE
[chore] add yarn-deduplicate to postinstall, fix root dependencies

### PR DESCRIPTION
**Pull request by Simek**
# Why & How

This small PR adds yarn deduplicate script, which has been hooked to the workspace `postinstall`. The deduplication is handled by `yarn-deduplicate` package from Atlassian. 

You can read more about what the script does and why it's important in the Readme section:
* https://github.com/atlassian/yarn-deduplicate#duplicated-packages

The changeset also includes the updated `yarn.lock` after the initial script run and a merge of workspace dependencies groups (because it doesn't matter in private and root workspace `package.json`s if dependency is defined as dev or normal one).

This script and the similar practice has been implemented in Docusaurus repository few months ago (https://github.com/facebook/docusaurus/blame/master/package.json#L64), and there were no issues with this cleanup across few released versions. It has been also added to the React Native scripts (https://github.com/facebook/react-native/blob/master/package.json#L69), but it's not hooked up into `postinstall` in there.

# Test Plan

I was able to install workspace, build few packages and run the docs website after changes to the lock.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
   "scripts": {
     "setup:docs": "./scripts/download-dependencies.sh",
     "setup:native": "./scripts/download-dependencies.sh && ./scripts/setup-react-android.sh",
-    "postinstall": "expo-yarn-workspaces check-workspace-dependencies",
+    "postinstall": "expo-yarn-workspaces check-workspace-dependencies && npx yarn-deduplicate",
```
Comment (brentvatne) [inline package.json:10]: i'd rather add this as a workspace dependency and update it as needed, instead of using npx

**Review by brentvatne — APPROVED** [review]
this is a nice change, cc @ide for sign off

File: package.json (hunk 1 above)
Comment (Simek) [reply package.json:10]: No problem, I will update the PR.

Comment (ide) [reply package.json:10]: Run deduplication before check-workspace-dependencies, which checks for correctness.

File: yarn.lock [hunk 2]
```diff
@@ -16346,6 +16356,39 @@ react-native-webview@11.2.3:
     escape-string-regexp "2.0.0"
     invariant "2.2.4"
 
+react-native@0.63.2:
```
Comment (ide) [inline yarn.lock]: I'm not sure react-native should ever show up in yarn.lock -- historically this has indicated issues with not using the symlinked copy from the submodule. You may need to run git submodule init, etc...

**Review by bbarthec — APPROVED** [review]

**Pull request merged**